lib/takeaway.rb

```diff
@@ -0,0 +1,64 @@
+require 'twilio-ruby'
+require_relative 'menu'
+require_relative 'sms'
+
+class Takeaway
+
+
+  attr_reader :menu, :order
+
+  def initialize(menu = Menu.new)
+    @menu = menu
+    @order = {}
+#    @sms = sms
+  end
+
+  def view_menu
+    menu.print_menu
+  end
+
+
+  # need to think about what place order will do at the moment it is responsible for sending the text!!!
+  def place_order
+    account_id
+    message
+  end
+
+
+  def add_to_order(dish, quantity)
+    fail "#{dish} is not on the menu" unless menu.dishes.has_key?(dish.downcase)
+    order[dish] = quantity
+  end
+
+  def total
+    order_totals.sum
+  end
+
+  private
+
+  def order_totals
+    order.map do |dish, quantity|
+      menu.dishes[dish.downcase] * quantity
+    end
+  end
+
+  def account_id
+    account_sid = "ACe16a4eddf4b5e19a3331831d7df88b9b"
+    auth_token = "f6964d00315fe133190e213d9cc775f1"
+    @client = Twilio::REST::Client.new(account_sid, auth_token)
```

Comment:
As I said above, storing the logic for sending messages buried here within the Takeaway class is going to make testing it correctly very difficult.

This line in particular is a bit of a red flag - within a method you have a call to .new on a different class in this case a Twilio Client object.

Creating new instances of objects like this is going to prevent you from properly isolating your unit tests. A better approach would be to make use of dependency injection as you have above.